### PR TITLE
Build RPM installer

### PR DESCRIPTION
**Pull request by barrowkwan**
This will allow us to create RPM installer for RedHat/CentOS.

### CODE REVIEW

File: installers/rpm/agent.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+SERVICE_NAME=${1:-gocd-golang-agent}
+
+if [ -f /etc/default/${SERVICE_NAME} ]; then
+  echo "[`date`] using default settings from /etc/default/${SERVICE_NAME}"
+  . /etc/default/${SERVICE_NAME}
+fi
+
+CWD=`dirname "$0"`
```
Comment (wpc) [inline installers/rpm/agent.sh:26]: is CWD used anywhere?

**Comment by wpc** [comment]
hi, @barrowkwan, the PR looks good. Does the rpm build script works with rpmbuild from homebrew on Mac or I need a centos/redhat machine to run it?

File: installers/rpm/agent.sh (hunk 1 above)
Comment (barrowkwan) [inline installers/rpm/agent.sh:26]: I believe eventually we will need it

**Comment by barrowkwan** [comment]
@wpc , you need centos/redhat to build the RPM ( also need the rpmbuild related package to be installed ).  I am working on two things right now.  1.  Vagrant to build CentOS/Redhat VM in VirtualBox.  2. build the rpm straight from OSX.

**Comment by ketan** [comment]
We build some rpms on snap -
https://github.com/gocd/nsis-rpm/blob/master/Rakefile

Comes complete with publishing a yum repository out of the box.

**Pull request merged**